src/commands/serve.rs

```diff
@@ -76,30 +77,53 @@ pub(crate) fn serve(args: &ArgMatches) {
 }
 
 fn serve_foreground(args: &ArgMatches) {
-    info!("Starting PHP...");
+    let port = find_available_port(parse_default_port(args.value_of("port").unwrap_or(DEFAULT_PORT), DEFAULT_PORT));
+
+    let mut document_root = get_document_root(args.value_of("document-root").unwrap_or("").to_string());
+    if document_root.ends_with('/') { document_root.pop(); }
+    if document_root.ends_with('\\') { document_root.pop(); }
+    document_root.push_str(if cfg!(target_family = "windows") { "\\" } else { "/" });
+    let doc_root_path = PathBuf::from(document_root.as_str());
+    let scripts = vec!["index.php", "app_dev.php", "app.php"];
+    let script_filename = if args.is_present("passthru") {
+        args.value_of("passthru").unwrap_or("index.php").to_string()
+    } else {
+        for script in scripts {
+            let php_entrypoint_path = doc_root_path.join(script);
+            if php_entrypoint_path.is_file() {
+                script.to_string();
+            }
+        }
+        "index.php".to_string()
+    };
+
+    let php_entrypoint_path = doc_root_path.join(script_filename.as_str());
+    let php_server = if !php_entrypoint_path.is_file() {
+        warn!("No PHP entrypoint file");
+        PhpServer::new(0, PhpServerSapi::Unknown)
+    } else {
+        info!("Starting PHP...");
+
+        php_server::start()
+    };
 
-    let php_server = php_server::start();
 
     let sapi = match php_server.sapi() {
         PhpServerSapi::FPM => "FPM",
         PhpServerSapi::CLI => "CLI",
         PhpServerSapi::CGI => "CGI",
         PhpServerSapi::Unknown => "?",
     };
-    info!("PHP started with module {}", sapi);
+    if sapi == "?" {
+        info!("Skip PHP start");
```

Comment:
If SAPI is unknown, we should panic and explain that we don't support anything else than FPM/CGI and CLI

Reply:
To implement this behavior and limit the change into the code, I have used the `Unknow` SAPI type and set the `php_port` value at 0.

If no PHP entry point is found, Rymfony start without PHP and **serve only static file**.

Reply:
> If no PHP entry point is found, Rymfony start without PHP and **serve only static file**.

This is not how I would have handled this error.

As said elsewhere, I prefer to have a true panic when unknown SAPI is detected and/or no entrypoint, because rendering static as default isn't the main point of Rymfony (which is a replacement for Symfony CLI).

I'm totally okay with keeping this in mind for long-term roadmap (so people could use Rymfony as a classic static files server only), but I prefer to stabilize the Symfony CLI related features beforehand.
In the future we might make `rymfony serve` an alias for `serve:php` for instance, and `rymfony serve:static` could be an interesting new way to use it.

In the meantime, `rymfony serve`'s behavior is clear: start a PHP server, and start a web server that proxies on it through FastCGI. Consistency 😅 .

Reply:
The SymfonyCli behavior

![image](https://user-images.githubusercontent.com/814683/105231086-98022000-5b66-11eb-9b42-9a0e31bd3cd5.png)